engines/dradis-api/spec/requests/dradis/ce/api/v1/attachments_spec.rb

```diff
@@ -0,0 +1,308 @@
+require 'spec_helper'
+
+describe "Attachments API" do
+
+  include_context "project scoped API"
+  include_context "https"
+
+  let(:node) { create(:node) }
+
+  context "as unauthenticated user" do
+    describe "GET /api/nodes/:node_id/attachments" do
+      it "throws 401" do
+        get "/api/nodes/#{node.id}/attachments", env: @env
+        expect(response.status).to eq 401
+      end
+    end
+    describe "GET /api/nodes/:node_id/attachments/:filename" do
+      it "throws 401" do
+        get "/api/nodes/#{node.id}/attachments/image.jpg", env: @env
+        expect(response.status).to eq 401
+      end
+    end
+    describe "POST /api/nodes/:node_id/attachments" do
+      it "throws 401" do
+        post "/api/nodes/#{node.id}/attachments", env: @env
+        expect(response.status).to eq 401
+      end
+    end
+    describe "PUT /api/nodes/:node_id/attachments/:filename" do
+      it "throws 401" do
+        put "/api/nodes/#{node.id}/attachments/image.jpg", env: @env
+        expect(response.status).to eq 401
+      end
+    end
+    describe "DELETE /api/nodes/:node_id/attachments/:filename" do
+      it "throws 401" do
+        delete "/api/nodes/#{node.id}/attachments/image.jpg", env: @env
+        expect(response.status).to eq 401
+      end
+    end
+  end
+
+  context "as authenticated user" do
+    include_context "authenticated API user"
+
+    after do
+      FileUtils.rm_rf(Attachment.pwd.join(node.id.to_s))
+    end
+
+    describe "GET /api/nodes/:node_id/attachments" do
+      before do
+        @attachments = ["image0.png", "image1.png", "image2.png"]
+        @attachments.each do |attachment|
+          create(:attachment, filename: attachment, node: node)
+        end
+
+        # an attachment in another node
+        create(:attachment, filename: "image3.png", node: create(:node))
+
+        get "/api/nodes/#{node.id}/attachments", env: @env
+      end
+
+      let(:retrieved_attachments) { JSON.parse(response.body) }
+
+      it "responds with HTTP code 200" do
+        expect(response.status).to eq(200)
+      end
+
+      it "retrieves all the attachments for the given node" do
+        expect(retrieved_attachments.count).to eq @attachments.count
+        retrieved_filenames = retrieved_attachments.map{ |json| json['filename'] }
+        expect(retrieved_filenames).to match_array(@attachments)
+      end
+
+      it "returns JSON information about attachments" do
+        attachment_0 = retrieved_attachments.detect { |n| n["filename"] == "image0.png" }
+        attachment_1 = retrieved_attachments.detect { |n| n["filename"] == "image1.png" }
+        attachment_2 = retrieved_attachments.detect { |n| n["filename"] == "image2.png" }
+
+        expect(attachment_0.keys).to match_array %w[filename link]
```

Comment:
I think it's enough to expect these hashes to be `eq` the expected hash (something like `expect(attachment_n).to eq(expected_hash)`), instead of asserting their keys and then asserting the values of each key. What do you think?